drivers/vulkan/vulkan_context.cpp

```diff
@@ -2468,7 +2468,7 @@ Error VulkanContext::prepare_buffers() {
 	return OK;
 }
 
-Error VulkanContext::swap_buffers() {
+Error VulkanContext::swap_buffers(bool p_present) {
```

Comment:
Conceptually, swapping buffers involves presenting. This would be a bit confusing. Wouldn't `flush(true, true)` would the right thing to call instead of `swap_buffers(false)` when there's no intention to present?

Reply:
Yeah I was thinking about splitting it or something. I'm going from the old situation where this existing function both flushes the command queue and does the swap + present, to it only doing the flush.
So maybe have a function called `flush` and then `swap_buffers`? It would clean up the change log a lot too as I don't need to indent all the code.

Reply:
I ended up just creating a separate method with just the flush of the buffers. Should look into if the existing `flush` method does a similar thing and shouldn't be used instead, I'm not sure enough about whats going on here

Reply:
Your new `flush_buffer()` function still detals present-related concerns. I'm certain that `flush(true, true)` will do the trick.

Reply:
Seems to work but I did get a small performance penalty, so I am worried about the device wait idle that we're introducing. Is it really required to wait for that or can we make that optional?

Reply:
Oh, those waits totally break CPU-GPU parallelism... Sorry for the bad idea. I'd still leverage `flush()`, though. What do you think about adding a third `bool p_sync = true` parameter. `vkDeviceWaitIdle()`s will happen only if `true`. For this usage of yours, you'd pass `false`, of course.

Reply:
Ah, that makes perfect sense, yeah I will add the `p_sync`, thats an easy fix

Reply:
Didn't seem to make a lot of difference but I'm still around the 9ms frame time with my example project.